cmd/operator/main.go

```diff
@@ -68,6 +68,12 @@ func main() {
 		webhookAddr = flag.String("webhook-addr", ":10250",
 			"Address to listen to for incoming kube admission webhook connections.")
 		metricsAddr = flag.String("metrics-addr", ":18080", "Address to emit metrics on.")
+
+		// Permit the operator to cleanup previously-managed resources that
+		// are missing the provided annotation. An empty string disables this
+		// feature.
+		cleanupAnnotKey = flag.String("cleanup-unless-annotation-key", "",
```

Comment:
If a resource doesn't have this annotation key, it will be deleted. e.g.
`--cleanup-unless-annotation-key=keep-my-pod` will delete any `collector` or `rule-evaluator` without that annotation key. The [unit tests](https://github.com/GoogleCloudPlatform/prometheus-engine/blob/pintohutch/annot-flag/pkg/operator/operator_test.go#L183) demonstrate how it works better I think.